### PR TITLE
fix: signed build issue with R8 full mode

### PR DESCRIPTION
**Pull request by farhan-arshad-dev**
## Description

[LEARNER-10122](https://2u-internal.atlassian.net/browse/LEARNER-10122)

- Enabled R8/Proguard with full mode.
- Make the signed build acceptable by the Google Play console by fixing the user Permission merging issue.

### CODE REVIEW

File: course/proguard-rules.pro [hunk 1]
```diff
@@ -1,21 +1,7 @@
...
-# hide the original source file name.
-#-renamesourcefileattribute SourceFile
\ No newline at end of file
+# Prevent shrinking, optimization, and obfuscation of the library when consumed by other modules.
+# This ensures that all classes and methods remain available for use by the consumer of the library.
+# Disabling these steps at the library level is important because the main app module will handle
+# shrinking, optimization, and obfuscation for the entire application, including this library.
+-dontshrink
+-dontoptimize
+-dontobfuscate
```
Comment (omerhabib26) [inline course/proguard-rules.pro:7]: please add a line at EOF

**Review by omerhabib26 — APPROVED** [review]
just a minor nit

**Review by dixidroid — APPROVED** [review]

**Review by HamzaIsrar12 — COMMENTED** [review]
I have two concerns:

- The `shrinkResources` is not set to `true`.
- The `mappingFileUploadEnabled` for Crashlytics is set to `false`.

Were these changes intentional?

**Comment by farhan-arshad-dev** [comment]
> I have two concerns:
> 
> * The `shrinkResources` is not set to `true`.
> * The `mappingFileUploadEnabled` for Crashlytics is set to `false`.
> 
> Were these changes intentional?

Will consider these as part of the release. For now, just pro-guard things.

File: course/proguard-rules.pro (hunk 1 above)
Comment (farhan-arshad-dev) [reply course/proguard-rules.pro:7]: can ignore it for now, will consider it as part of upstream.

**Pull request merged**